lxd/db/query/objects.go

```diff
@@ -31,6 +31,31 @@ func SelectObjects(stmt *sql.Stmt, dest Dest, args ...any) error {
 	return nil
 }
 
+// QueryObjects executes a query from a SQL string which must yield rows with
+// a specific columns schema. It invokes the given Dest hook for each yielded row.
+func QueryObjects(tx *sql.Tx, stmt string, dest Dest, args ...any) error {
+	rows, err := tx.Query(stmt, args...)
+	if err != nil {
+		return err
+	}
+
+	defer func() { _ = rows.Close() }()
+
+	for i := 0; rows.Next(); i++ {
+		err := rows.Scan(dest(i)...)
```

Comment:
Not a blocker, just a nit, but I would caution against creating a new scoped `err` here and instead just re-use the existing one by doing `err = ...` instead.